File: proyectos_c#/web/DemoPruebasUnitarias/DemoPruebasUnitarias.Tests/DivisionTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace DemoPruebasUnitarias.Tests
{
    [TestClass]
    public class DivisionTests
    {
        [TestMethod]
        [ExpectedException(typeof(DivideByZeroException))]
        public void divisionDivideEntreCero()
        {
            //Arrange o Planteamiento
            const int dividendo = 8;
            const int divisor = 0;
            const int esperado = 8;
            //Act o Prueba
            var actual = Matematicas.division(dividendo, divisor);

            //Assert o Excepción
            Assert.AreEqual(esperado, actual);
        }
    }
}

File: proyectos_basic/web/ProyectoUnitTest/ProyectoUnitTest.Tests/DivisionTests.vb
Imports System.Text
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass()> Public Class DivisionTests

    <TestMethod()>
    <ExpectedException(TypeOf (DivideByZeroException))>
    Public Sub divisionDivideEntreCero()

        'Arrange o Planteamiento
        Const dividendo As Integer = 8
        Const divisor As Integer = 0
        Const esperado As Integer = 8
        'Act o Prueba
        Dim actual As Integer = Matematicas.division(dividendo, divisor)

        'Assert o Excepción
        Assert.AreEqual(esperado, actual)
    End Sub
End Class